src/Form/MessageSettingsForm.php

```diff
@@ -100,21 +100,17 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ],
     ];
 
-    $form['purge']['purge_quota'] = [
-      '#type' => 'textfield',
-      '#title' => t('Messages quota'),
-      '#description' => t('Maximal (approximate) amount of messages.'),
-      '#default_value' => $config->get('purge_quota'),
-      '#states' => $states,
-    ];
+    /** @var \Drupal\message\MessagePurgeInterface $plugin */
+    foreach (array_keys(\Drupal::service('plugin.manager.message.purge')->getDefinitions()) as $plugin_id) {
+      $plugin = \Drupal::service('plugin.manager.message.purge')->createInstance($plugin_id);
+      $element = $plugin->configurationForm($form, $form_state);
+      $element['#states'] = $states;
 
-    $form['purge']['purge_days'] = [
-      '#type' => 'textfield',
-      '#title' => t('Purge messages older than'),
-      '#description' => t('Maximal message age in days.'),
-      '#default_value' => $config->get('purge_quota'),
-      '#states' => $states,
-    ];
+      // @todo: Set the default value.
+      // $element['#default_value'] = $config->get($plugin_id);
```

Comment:
Ok, Tim Plunket gave me the answer - It's very similar to the the image style plugins per plugin config.